### PR TITLE
Delete `dataset-split-names-from-dataset-info` job runner

### PR DESCRIPTION
**Pull request by polinaeterna**
Part of https://github.com/huggingface/datasets-server/issues/1086

### CODE REVIEW

File: services/api/tests/routes/test_endpoint.py [hunk 1]
```diff
@@ -36,7 +36,7 @@ def test_endpoints_definition() -> None:
     assert sorted(list(splits)) == ["config", "dataset"]
     assert splits["dataset"] is not None
     assert splits["config"] is not None
-    assert len(splits["dataset"]) == 2  # Has two processing steps
+    assert len(splits["dataset"]) == 1  # Has one processing steps
```
Comment (AndreaFrancis) [inline services/api/tests/routes/test_endpoint.py:39]:
```suggestion
    assert len(splits["dataset"]) == 1  # Has one processing step
```

**Review by AndreaFrancis — APPROVED** [review]
Very nice! delete_migration.py will be really helpful

File: jobs/mongodb_migration/tests/migrations/test_20230504185100_cache_delete_dataset_split_names_from_dataset_info.py
Comment (severo) [inline jobs/mongodb_migration/tests/migrations/test_20230504185100_cache_delete_dataset_split_names_from_dataset_info.py:1]: I think we don't need a specific test, now that we use DeleteCacheMigration. This method should be tested instead.

File: jobs/mongodb_migration/tests/migrations/test_20230504194600_metrics_delete_dataset_split_names_from_dataset_info.py
Comment (severo) [inline jobs/mongodb_migration/tests/migrations/test_20230504194600_metrics_delete_dataset_split_names_from_dataset_info.py:1]: Same: should be a test on DeleteMetricsMigration

**Review by severo — APPROVED** [review]
Nice improvement. If you can, please change the tests from specific to generic. Can be in another PR too

**Pull request merged**